Content.Server/GameObjects/Components/StationEvents/RadiationPulseComponent.cs

```diff
@@ -18,48 +18,68 @@ public sealed class RadiationPulseComponent : SharedRadiationPulseComponent
+        ///     <see cref="MinPulseLifespan"/> and <see cref="MaxPulseLifespan"/>
+        /// </summary>
+        public bool Decay { get; set; }
+        public float MinPulseLifespan { get; set; }
+        public float MaxPulseLifespan { get; set; }
+        public float DPS { get; set; }
+        public string Sound { get; set; }
 
-        public float DPS => _dps;
-        private float _dps;
+        /// <summary>
+        /// Radius of the pulse from its position
+        /// </summary>
+        public float Range { get; set; }
 
+        public bool Draw { get; set; }
+
+        private float _duration;
         private TimeSpan _endTime;
 
         public override void ExposeData(ObjectSerializer serializer)
         {
             base.ExposeData(serializer);
-            serializer.DataField(ref _dps, "dps", 40.0f);
+            serializer.DataField(this, x => x.DPS, "dps", 40.0f);
+            serializer.DataField(this, x => x.Sound, "sound", "/Audio/Weapons/Guns/Gunshots/laser3.ogg");
+            serializer.DataField(this, x => x.Range, "range", 5.0f);
+            serializer.DataField(this, x => x.Draw, "draw", true);
+            serializer.DataField(this, x => x.Decay, "decay", true);
+            serializer.DataField(this, x => x.MaxPulseLifespan, "maxPulseLifespan", 2.5f);
+            serializer.DataField(this, x => x.MinPulseLifespan, "minPulseLifespan", 0.8f);
         }
 
-        public override void Initialize()
+        public void DoPulse()
         {
-            base.Initialize();
-
-            var currentTime = _gameTiming.CurTime;
-            var duration  =
-                TimeSpan.FromSeconds(
-                    _random.NextFloat() * (MaxPulseLifespan - MinPulseLifespan) +
-                    MinPulseLifespan);
-
-            _endTime = currentTime + duration;
-
-            Timer.Spawn(duration,
-                () =>
+            if (Decay)
             {
-                if (!Owner.Deleted)
-                {
-                    Owner.Delete();
-                }
-            });
+                var currentTime = _gameTiming.CurTime;
+                _duration = _random.NextFloat() * (MaxPulseLifespan - MinPulseLifespan) + MinPulseLifespan;
+                _endTime = currentTime + TimeSpan.FromSeconds(_duration);
+            }
+
+            if(!string.IsNullOrEmpty(Sound))
+                EntitySystem.Get<AudioSystem>().PlayAtCoords(Sound, Owner.Transform.GridPosition);
 
-            EntitySystem.Get<AudioSystem>().PlayAtCoords("/Audio/Weapons/Guns/Gunshots/laser3.ogg", Owner.Transform.GridPosition);
-            Dirty();
+            if(Draw)
+                Dirty();
         }
 
         public override ComponentState GetComponentState()
         {
-            return new RadiationPulseMessage(_endTime);
+            return new RadiationPulseState(_endTime, Range);
+        }
+
+        public void Update(float frameTime)
+        {
+            if (!Decay || !Owner.Deleted)
```

Comment:
Should this be Owner.Deleted or am I misreading it

Reply:
You're right, fixed now